inspirehep/modules/disambiguation/config.py

```diff
@@ -25,5 +25,11 @@
 from __future__ import absolute_import, division, print_function
 
 
+DISAMBIGUATION_CLUSTERS_PATH = '/tmp/clusters.json'
+"""The path to the file that will store the curated clusters."""
+
+DISAMBIGUATION_SIGNATURES_PATH = '/tmp/signatures.jl'
```

Comment:
Is the file extension correct here?

Reply:
Yes, it's a less used extension to mean this format: http://jsonlines.org/

Actually that's another breaking change from the paper: in order to write a big JSON array to disk one would need to first have it in memory, and I believe that no INSPIRE machine has enough RAM to do that.

If instead we save each signature as a JSON line we can write to disk as soon as we read a signature from the DB.

Reply:
All it means is that code like
```python
with open('signatures.json') as fd:
    signatures = json.load(fd)
    for signature in signatures:
        ...
```
becomes
```python
with open('signatures.jl') as fd:
    for line in fd:
        signature = json.loads(line)
        ...
```

Reply:
I checked the size of the data in memory (using sys.getsizeof ) provided by the beard paper repo and it's barely a few MBs.

Reply:
> I checked the size of the data in memory (using sys.getsizeof ) provided by the beard paper repo and it's barely a few MBs.

Can you detail exactly what you are doing, so we know that we're comparing apples to apples? The signatures that we produce are comparable to `signatures.json` + `records.json`, and we can expect a lot more signatures and higher record metadata quality (in particular more abstracts, since they couldn't release the publisher ones, but we are free to use them while training!)

Reply:
htop and ctop give a different picture though. At least 3 GB in memory. I suppose you can go ahead with your proposal. It's better anyway.

Reply:
> Can you detail exactly what you are doing

So I loaded the signatures.json and records.json in a Jupyter notebook using json.load. Then I checked using sys.getsizeof on both records and signatures and they return ~ 9 MB and ~ 3 MB respectively. However, when I checked the memory usage in htop, it increased by ~ 3 GB. I might not be doing something correct using getsizeof.

Reply:
`sys.getsizeof` only looks at the size of the top-level object in case of collections:
```
In [1]: import sys

In [2]: d = {'foo': {}}

In [3]: sys.getsizeof(d)
Out[3]: 280

In [4]: d['foo'] = {i: i for i in xrange(100000)}

In [5]: sys.getsizeof(d)
Out[5]: 280

In [6]: sys.getsizeof(d['foo'])
Out[6]: 6291736

```

Reply:
Thanks for the explanation. That makes sense.